app.js

```diff
@@ -1,8 +1,99 @@
-// import functions and grab DOM elements
+import { renderGoblins } from './render-utils.js';
+
+const goblinStableEl = document.querySelector('.goblin-stable');
+const goblinsDefeatedEl = document.querySelector('.goblins-defeated');
+const playerAvatarEl = document.querySelector('.player-avatar');
+
+const form = document.querySelector('form');
+
+const playerHPEl = document.querySelector('.player-hp');
+
+
+let goblins = [
+    { name: 'Goblin 1', hp: 2, },
+    { name: 'Goblin 2', hp: 3, },
+    { name: 'Goblin 3', hp: 4, },
+];
+
+let playerHP = 10;
+
+let goblinsDefeated = 0;
+
+displayGoblins();
+
+form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const data = new FormData(form);
+
+    const newGoblinName = data.get('new-goblin-name');
+
+    const newGoblin = { name: newGoblinName, hp: Math.floor(Math.random() * 6) + 3 };
+
+    goblins.push(newGoblin);
+
+    form.reset();
+    displayGoblins();
+});
+
+function displayGoblins() {
+
+    goblinStableEl.textContent = '';
+    for (let goblin of goblins) {
+
+        const goblinEl = renderGoblins(goblin);
+
+        let emoji = '👺';
+        if (goblin.hp <= 0) {
+            emoji = '🔥';
+        }
+
+        goblinEl.textContent = `${goblin.name} ${emoji} ${goblin.hp} HP`;
```

Comment:
Consider using an expression directly inside the template literal to set this emoji. You can use something called a `ternary operator` as a replacement for those short-and-sweet if/else statements.  It may or may not improve readability, so it's up to you whether you'd like to implement something like this...

```js
goblinEl.textContent = `${goblin.name} ${(goblin.hp <= 0) ? '🔥' : '👺'} ${goblin.hp} HP`;
```